_blog-src/_posts/2016-04-08-data-encryption.adoc

```diff
@@ -0,0 +1,142 @@
+
+We often forget that APIs are not meant to protect data, rather, to share data across applications. While providers are certainly responsible for designing their APIs to be secure, we must be equally cautious about _each_ component of the API layer.
+
+=== Data Storage in apiman
+
+Designing, building, and maintaining an API can be a lot of work, often making security an afterthought. Using an API management platform can save you valuable time, allowing you to focus on other critical aspects of your business.
+
+Apiman was designed with the goal of providing a secure, customizable platform for API providers that is easy to use. As an open source platform, it is highly extendable and maintained by a community of developers with the common goal of streamlining API management without compromising security.
+
+*All sensitive data stored by apiman is fully encrypted* so that it is not stored in plain text, then later decrypted again on runtime or when publishing services. By default, the database storage is the H2 relational database, which uses Elasticsearch. One of the many examples of data that gets stored are the credentials that must be provided when the manager wants to publish an API and makes an authenticated REST call to the gateway.
+
+To provide an additional layer of protection we also leverage an object relational mapper built with JPA (Java Persistence API), the specification for relational data in Java-based applications. The framework that implements this specification is called Hibernate, an established and popular ORM in the Java community.
+
+Data for the API gateway and API manager are stored separately to keep components fully decoupled.
+
+For a more detailed overview, please see our http://www.apiman.io/latest/production-guide.html#_architecture_summary[API layer architecture]. Let’s swan dive into some code!
+
+=== IDataEncrypter Interface and Data Encryption Classes
+
+The interface that is used for all policies is called `IDataEncrypter`. This interface provides the structure for the classes that encrypt keys and data, which is done synchronously using AES.
+
+By default, the `DefaultDataEncrypter` class is the encrypter for the gateway. The encrypt and decrypt methods accept a single parameter for the string to be encrypted or decrypted, respectively. If you look at the code, you'll notice that the key is hard coded. For those of us that are looking for a more secure approach, we offer an alternative.
+
+The `AesDataEncrypter` class is a data encrypter that accepts an encrypted secret key. When it is invoked, it expects a configuration object that contains a previously defined encrypted secret key.
+
+At this point you may be wondering how secret keys are obtained. A factory class called `ConfigFactory` is responsible for creating a configuration object using `apiman.properties`. This object is later passed to `AesDataEncrypter` as a parameter, along with the string of text to be encrypted or decrypted.
```

Comment:
Suggested change:

At this point you may be wondering how secret keys are obtained. A factory class called `ConfigFactory` is responsible for creating a configuration Map using information from `apiman.properties`. This configuration Map is later passed to `AesDataEncrypter` as a parameter when it is constructed.  The secret key included in the configuration Map is then used whenever data is encrypted or decrypted.